webpack/InsightsCloudSync/Components/InsightsTable/InsightsTableSelectors.js

```diff
@@ -0,0 +1,74 @@
+import URI from 'urijs';
+import { STATUS } from 'foremanReact/constants';
+import { selectRouterLocation } from 'foremanReact/routes/RouterSelector';
+import {
+  selectAPIResponse,
+  selectAPIStatus,
+  selectAPIError,
+} from 'foremanReact/redux/API/APISelectors';
+import { INSIGHTS_HITS_API_KEY } from './InsightsTableConstants';
+import { selectInsightsCloudSync } from '../../../ForemanRhCloudSelectors';
+
+export const selectQuery = state => selectRouterLocation(state).query;
+
+export const selectSearch = state => {
+  const { search = '' } = selectQuery(state);
+  return URI.decodeQuery(search);
+};
+
+export const selectPage = state => {
+  const { page = '1' } = selectQuery(state);
+  return Number(page);
+};
+
+export const selectPerPage = state => {
+  const perPage = selectQuery(state).per_page;
+  return perPage && Number(perPage);
+};
+
+export const selectQueryParams = state => ({
+  page: selectPage(state),
+  perPage: selectPerPage(state),
+  query: selectSearch(state),
+  sortBy: selectSortBy(state),
+  sortOrder: selectSortOrder(state),
+});
+
+export const selectStatus = state =>
+  selectAPIStatus(state, INSIGHTS_HITS_API_KEY);
+
+export const selectError = state =>
+  selectAPIError(state, INSIGHTS_HITS_API_KEY);
```

Comment:
Last issue: I think this should be `selectAPIErrorMessage` as `selectAPIError` returns an object and not a string so I get this warning:
```
Warning: Failed prop type: Invalid prop `error` of type `object` supplied to `InsightsTable`, expected `string`.
```